pwndbg/exception.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import pdb
+import sys
+import traceback
+
+import pwndbg.config
+import pwndbg.stdio
+
+try:
+	import ipdb as pdb
+except ImportError:
+	pass
+
+verbose = pwndbg.config.Parameter('exception-verbose', False, 'whether to print a full stacktracefor exceptions raised in Pwndbg commands')
+debug = pwndbg.config.Parameter('exception-debugger', False, 'whether to debug exceptions raised in Pwndbg commands')
+
+def handle():
+    # Display the error
+    if debug or verbose:
+        print(traceback.format_exc())
+    else:
+        print(sys.exc_info()[1])
```

Comment:
Just for the weird malloc case it says:
```
evaluation of this expression requires the program to have a function "malloc".
```

Might be good to do:
```python
print("{}: {}".format(*sys.exc_info())
```
This would produce:
```
<class 'gdb.error'>: evaluation of this expression requires the program to have a function "malloc".
```

However... in cases like #249 this is the output (with the `.format` stuff):
```
<class 'gdb.error'>: evaluation of this expression requires the program to have a function "malloc".
'context': Print out the current register, instruction, and stack context.

    Accepts subcommands 'reg', 'disasm', 'code', 'stack', 'backtrace', and 'args'.
<class 'TypeError'>: object of type 'NoneType' has no len()
```

Probably it would be better to do one of those:

1. Just inform the user...
```python
print("Pwndbg error. If you want to see full stracktrace, fire `set exception-verbose on` and redo the thing you have done. Report the issue on https://github.com/pwndbg/pwndbg/issues")
```

2. Save the exception in a global state and let the user see it by either `py print(pwndbg.exception.last)` or `pwndbg-error` command.

3. Save the exception and let the user generate issue/bug report through a command.